### DISCUSSION TITLE
Support pgx as a postgres driver

### DISCUSSION THREAD

**Discussion by crazybolillo** [amacneil/dbmate#595] [Feature Ideas]
I mostly use pgx in my projects, however to run migrations as a library with dbmate, I must use the postgres driver package (as far as I understand) which uses libpq. This means I have to pull two different SQL drivers.

Is it possible to use pgx instead? Or maybe add a new package postgres-pgx which pulls in pgx instead of libqp?

**Comment by crazybolillo** [comment]
Also, if this is accepted, I am willing to work on a PR for it.

**Comment by dossy** [comment]
[Postgres's `libpq`](https://www.postgresql.org/docs/current/libpq.html) library is a C library, but `dbmate` uses the [Go `pq`](https://github.com/lib/pq) package, which could be called `lib/pq` and may lead to confusion with the C library `libpq`, but they are not the same and the Go `pq` package is a pure Go implementation and does not depend on the Postgres `libpq` C library.
